apps/prairielearn/src/pages/instructorFileEditor/instructorFileEditor.js

```diff
@@ -283,18 +295,15 @@ router.post('/*', (req, res, next) => {
         async () => {
           debug('Write edit to db');
           await createEdit(fileEdit);
-        },
-        (callback) => {
+
           debug('Write edit to disk (also push and sync if necessary)');
           fileEdit.needToSync = path.extname(fileEdit.fileName) === '.json';
-          saveAndSync(fileEdit, res.locals, (err) => {
-            // If there is an error, log it and pass null to the callback.
-            if (ERR(err, (err) => logger.info(err))) {
-              callback(null);
-              return;
-            }
-            callback(null);
-          });
+          try {
+            await saveAndSync(fileEdit, res.locals);
+          } catch (err) {
+            // If there is an error, log it and carry on.
+            logger.error('Error saving and syncing instructor file edit', err);
+          }
```

Comment:
Error handling is necessary because `serverJob.execute` will propagate errors back to the caller. This behavior was introduced in https://github.com/PrairieLearn/PrairieLearn/pull/7949, but in this case, I think that's actually undesirable. Maybe we need another variant of `execute` like `executeSafe`? Or maybe `execute` and `executeUnsafe`?

Reply:
The use case for `executeUnsafe` is actually the *other* editing machinery:

https://github.com/PrairieLearn/PrairieLearn/blob/41bd6187f6000d0deb11baec7647bd411fa76919/apps/prairielearn/src/lib/editors.js#L113

The existing code relies on an error being propagated back upwards so that they can end up at the `.../edit_error/...` page.

Reply:
Ah! Ok, that makes sense. In that case I'm certainly on board with both a safe and unsafe version of `execute`. I'm not sure which should be the default -- it may even be worth having only `executeSafe()` and `executeUnsafe()` to force callers to explicitly decide. Is the safe/unsafe naming here conventional to indicate whether a function might throw an exception? I don't have a better solution but I do have a very slight concern that the unsafe variant sounds too scary to use, although maybe that's exactly what we want 😄

Reply:
I landed on `execute` and `executeUnsafe` in 17e7656 and removed redundant error handling in 9a34171. For server jobs, I think "don't propagate errors" is a reasonable default.

I don't know _execatly_ how conventiontional this naming is, but there's precedent in e.g. Zod, which uses `parse` and `safeParse`.